### PR TITLE
Selectv writelock

### PR DESCRIPTION
**Pull request by markhannum**
Grab a writelock for a selectv'd record that will later be modified.

### CODE REVIEW

File: db/db_tunables.c [hunk 1]
```diff
@@ -290,6 +290,8 @@ extern int gbl_osql_check_replicant_numops;
 extern int gbl_abort_on_missing_osql_session;
 extern int gbl_abort_irregular_set_durable_lsn;
 extern int gbl_legacy_schema;
+extern int gbl_selectv_writelock_on_update;
+extern int gbl_selectv_writelock;
```
Comment (adizaimi) [inline db/db_tunables.c]: This one seems to be a duplicate so can be deleted.

File: db/osqlcomm.c [hunk 2]
```diff
@@ -7487,7 +7503,10 @@ int osql_process_packet(struct ireq *iq, unsigned long long rqid, uuid_t uuid,
 
         lclgenid = bdb_genid_to_host_order(dt.genid);
 
-        rc = ix_check_genid(iq, trans, dt.genid, &bdberr);
+        if (gbl_selectv_writelock)
```
Comment (adizaimi) [inline db/osqlcomm.c]: Using wrong variable which is never set. Why does the testcase work?

File: tests/selectv_deadlock.test/lrl.options [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+maxosqltransfer 1000000
+enable_new_snapshot
+enable_snapshot_isolation
+berkattr abort_on_replicant_log_write 1
+init_with_genid48
+
+#reorder_socksql_no_deadlock on
+#reorder_idx_writes on
+
+#selectv_writelock_on_update is enabled by default
```
Comment (adizaimi) [inline tests/selectv_deadlock.test/lrl.options]: Clean up some of this?

File: db/db_tunables.c (hunk 1 above)
Comment (markhannum) [reply db/db_tunables.c]: No - selectv_writelock acquires a writelock for every selectv, selectv_writelock_on_update only acquires a writelock for rows that are updated later

**Pull request merged**